<launch>
    <arg name="robot1_active" default="true"/>
    <arg name="robot2_active" default="true"/>
    
    <group ns="robot1" if="$(arg robot1_active)">
        <node name="vive_trackerpose" pkg="vive" type="vive_trackerpose" output="screen">
            <rosparam file="$(find vive)/param/vive_trackerpose.yaml" command="load" />
            <param name="tracker" type="string" value="trackerA"/>
            <param name="covariance0" type="double" value="1.0"/>
            <param name="covariance7" type="double" value="1.0"/>
            <param name="covariance14" type="double" value="1.0"/>
            <param name="covariance21" type="double" value="1.0"/>
            <param name="covariance28" type="double" value="1.0"/>
            <param name="covariance35" type="double" value="1.0"/>
        </node>
    </group>
    <group ns="robot2" if="$(arg robot2_active)">
        <node name="vive_trackerpose" pkg="vive" type="vive_trackerpose" output="screen">
            <rosparam file="$(find vive)/param/vive_trackerpose.yaml" command="load" />
            <param name="tracker" type="string" value="trackerB"/>
            <param name="covariance0" type="double" value="1.0"/>
            <param name="covariance7" type="double" value="1.0"/>
            <param name="covariance14" type="double" value="1.0"/>
            <param name="covariance21" type="double" value="1.0"/>
            <param name="covariance28" type="double" value="1.0"/>
            <param name="covariance35" type="double" value="1.0"/>
        </node>
    </group>

</launch>